packages/compass-crud/src/stores/crud-store.spec.js

```diff
@@ -20,62 +20,120 @@ const CONNECTION = new Connection({
   mongodb_database_name: 'admin'
 });
 
-function listenToStore(store, cb, expectedCalls = 1) {
-  return new Promise((resolve, reject) => {
-    let numCalls = 0;
-    const unsubscribe = store.listen((state) => {
-      ++numCalls;
+function delay(timeout) {
```

Comment:
I have written this function so many times but I cannot bring myself to make a leftpad package for it.

Reply:
It exists, but only in node 15+ https://nodejs.org/api/timers.html#timerspromisessettimeoutdelay-value-options

Reply:
There’s also `util.promisify(setTimeout)`, fwiw, which does the same thing (but obviously still requires some extra code for importing util).

Reply:
Woah, didn't know that `setTimeout` implements a custom handler for promisify, neat!

Reply:
I have such déjà vu right now. Like I've belatedly discovered this before..